### PR TITLE
feat(discord): Implement Discord interactions endpoint

### LINKED ISSUE #50864: Implement Discord interactions endpoint
This endpoint is how Discord sends events to us. This needs to be able to handle generic events, identify them, and forward them to an appropriate handler.

This endpoint also needs verify request signatures from Discord. You can verify this is working by trying to set this endpoint as the 'interactions' endpoint for a Discord bot.

See [here](https://discord.com/developers/docs/tutorials/upgrading-to-application-commands#adding-an-interactions-endpoint-url) for more details.

### PR DESCRIPTION
**Pull request by spalmurray**
Implement minimal interactions endpoint that is able to receive, validate, and respond to PING events from Discord.

### CODE REVIEW

File: src/sentry/integrations/discord/requests/base.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+class DiscordRequestError(Exception):
+    """
+    Something was invalid about the request from Discord.
+    Includes the status the endpoint should return, based on the error.
+    """
+
+    status: int
+
+
+class DiscordRequest:
```
Comment (spalmurray) [inline src/sentry/integrations/discord/requests/base.py:25]: A lot of this PR is inspired by Slack's implementation. I liked how everything was split up there.

https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/slack/webhooks/event.py
https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/slack/requests/base.py

File: src/sentry/integrations/discord/utils/auth.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PublicKey
+
+
+def verify_signature(public_key_string: str, signature: str, message: str) -> bool:
+    try:
```
Comment (spalmurray) [inline src/sentry/integrations/discord/utils/auth.py:6]: Discord's docs recommended a different package for this, but I was able to figure out how to do this with our existing cryptography package! 😄

File: src/sentry/integrations/discord/requests/base.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    @property
+    def guild_id(self) -> str:
+        return str(self.data.get("guild_id", None))
+
+    @property
+    def logging_data(self) -> Mapping[str, str | int]:
+        _data: Mapping[str, object] = self.request.data
+        # TODO: come back to this later and see what additional metadata makes sense to include here
+        data: dict[str, str | int | None] = {
+            "discord_guild_id": str(_data.get("guild_id", None)),
```
Comment (snigdhas) [inline src/sentry/integrations/discord/requests/base.py:59]: I think `str(None)` evaluates to `'None'`. Is that what we want here?

Comment (spalmurray) [reply src/sentry/integrations/discord/requests/base.py:59]: Good catch. Slack does `None` not `'None'`, so I'll go with that.

File: src/sentry/integrations/discord/utils/auth.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+        # https://cryptography.io/en/latest/hazmat/primitives/asymmetric/ed25519/#cryptography.hazmat.primitives.asymmetric.ed25519.Ed25519PublicKey
```
Comment (snigdhas) [inline src/sentry/integrations/discord/utils/auth.py:7]: Is there something specific in the docs you'd like to reference here? If not, future users can simply look up the library docs. 

I'd add a sentence with the info that's important in this case.

**Review by snigdhas — APPROVED** [review]

File: src/sentry/integrations/discord/utils/auth.py (hunk 4 above)
Comment (spalmurray) [reply src/sentry/integrations/discord/utils/auth.py:7]: That is a good point. I suppose this link really doesn't add anything!

**Pull request merged**